addons/website_event/models/event_event.py

```diff
@@ -190,6 +211,51 @@ def _compute_website_url(self):
             if event.id:  # avoid to perform a slug on a not yet saved record in case of an onchange.
                 event.website_url = '/event/%s' % slug(event)
 
+    @api.depends('event_type_id')
+    def _compute_question_ids(self):
+        """ Update event questions from its event type. Depends are set only on
+        event_type_id itself to emulate an onchange. Changing event type content
+        itself should not trigger this method.
+
+        When synchronizing questions:
+
+          * lines that no answer are removed;
+          * type lines are added;
+        """
+        if self._origin.question_ids:
+            # lines to keep: those with already given answers
+            questions_tokeep_ids = self.env['event.registration.answer'].search(
+                [('question_id', 'in', self._origin.question_ids.ids)]
+            ).question_id.ids
+        else:
+            questions_tokeep_ids = []
+        for event in self:
+            if not event.event_type_id and not event.question_ids:
+                event.question_ids = self._default_question_ids()
```

Comment:
Note that this is not really in the original code being moved ... changing the default behavior should be done in its own commit, to ease traceability.